File: src/main/java/com/skocur/imagecipher/plugin/PluginManager.java
package com.skocur.imagecipher.plugin;

import java.util.LinkedList;
import java.util.List;
import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;

public class PluginManager {

  static final String PLUGINS_PATH = "plugins/";

  private static List<PluginConfiguration> plugins = new LinkedList<>();

  private static final Logger logger = LogManager.getLogger();

  public static void initialize() {
    logger.info("Initializing PluginLoader");

    PluginLoader pluginLoader = new PluginLoader();
    pluginLoader.loadAllPlugins(PluginManager.PLUGINS_PATH);
  }

  public static List<PluginConfiguration> getPlugins() {
    return plugins;
  }
}

File: app/src/main/kotlin/com/imagecipher/app/plugin/PluginManager.kt
package com.imagecipher.app.plugin

import org.apache.logging.log4j.LogManager

object PluginManager {

    const val PLUGINS_PATH = "plugins/"

    private val plugins: MutableList<PluginConfiguration> = mutableListOf()

    private val logger = LogManager.getLogger()

    fun initialize() {
        logger.info("Initializing PluginLoader")

        val pluginLoader = PluginLoader()
        pluginLoader.loadAllPlugins(PLUGINS_PATH)
    }

    fun getPlugins(): MutableList<PluginConfiguration> {
        return plugins
    }
}
